Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,568 @@
+```
+
+If we don't want to pass the query option in the request, but we want the `Orders` property to be expanded, we can initialize the `SelectExpandQueryOption` and set it in the request.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    options.Request.ODataFeature().SelectExpandClause = new SelectExpandQueryOption(null, "Orders", options.Context,
+        new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$expand", "Orders" } },
+            options.Context.RequestContainer)).SelectExpandClause;
+
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+When making the request, we don't need to use the `$expand` query option.
+
+```http
+GET http://localhost:6285/odata/Customers
+```
+
+### Filter query
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+In the example above, `options.Filter` property will be set when we make the request below:
+
+```http
+GET http://localhost:6285/odata/Customers?$filter=Id eq 1
+```
+The `options.Filter` property is an instance of `FilterQueryOption`.
+
+Below will be the output
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1",
+            "Age": 31
+        }
+    ]
+}
+```
+
+Alternatively we can call `ApplyTo` on the `Filter` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.Filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+If we don't want to pass the filter query option in the request, but we want a filter to be applied, we can initialize the `FilterQueryOption` and call `ApplyTo` to apply the filter query to the data.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    var filter = new FilterQueryOption("Id eq 1", options.Context,
+        new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$filter", "Id eq 1" } },
+            options.Context.RequestContainer));
+
+    IQueryable results = filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+When making the request, we don't need to use the `$expand` query option.
+
+```http
+GET http://localhost:6285/odata/Customers
+```
+
+### Apply query
```

Comment:
Because this is the first place you bring up the `Apply` query feature, maybe you should mention it's applied in grouping and aggregating data...